Userland/Libraries/LibEDID/EDID.cpp

```diff
@@ -382,19 +442,40 @@ ErrorOr<IterationDecision> Parser::for_each_extension_block(Function<IterationDe
         u8 tag = read_host(&raw_block.tag);
 
         if (current_extension_map && raw_index == 127) {
-            if (tag != (u8)Definitions::ExtensionBlockTag::ExtensionBlockMap)
+            if (tag != (u8)Definitions::ExtensionBlockTag::ExtensionBlockMap) {
+#ifndef KERNEL
                 return Error::from_string_literal("Did not find extension map at block 128");
+#else
+                return Error::from_errno(EINVAL);
+#endif
```

Comment:
All the changes in this file make it feel like we should invent another static API for Error. something like...

```
static Error from_string_with_errno(StringView error_message, ErrnoVal value)
{
#ifdef KERNEL
     dmesgln("{}", error_message);
     return Error::from_errno(value);
#else
    return Error::from_syscall(error_message, value);
#endif
}
```

Reply:
I really agree with this! I'll fix it now, thanks for the suggestion.